CHANGELOG.next.toml

```diff
@@ -10,6 +10,15 @@
 # references = ["smithy-rs#920"]
 # meta = { "breaking" = false, "tada" = false, "bug" = false, "target" = "client | server | all"}
 # author = "rcoh"
+
+[[aws-sdk-rust]]
+message = """
+Content-Type header validation now ignores parameter portion of media types.
+"""
+references = ["smithy-rs#3576"]
```

Comment:
@drganjoo found this old issue; apparently someone had identified this bug previously and we didn't notice because they adjusted the client-side! We should link it too: https://github.com/smithy-lang/smithy-rs/issues/3471

Reply:
I've added this and the tests requested above. I would still need to upstream to smithy-rs, but figured I'd get them here first and these can be the model for upstream.